plugins/tekton/src/components/pipeline-topology/PipelineVisualization.test.tsx

```diff
@@ -42,7 +48,40 @@ describe('PipelineVisualization', () => {
       screen.getByText(mockKubernetesPlrResponse.pipelineruns[1].metadata.name),
     ).toBeInTheDocument();
 
-    expect(queryByTestId('pipeline-visualization')).not.toBeNull();
+    expect(queryByTestId('latest-pipelineRun-visualization')).not.toBeNull();
+  });
+
+  it('should render the pipeline run visualization when pipelineRun name exists in the params', async () => {
+    const mockContextData = {
+      watchResourcesData: {
+        pipelineruns: {
+          data: mockKubernetesPlrResponse.pipelineruns,
+        },
+        taskruns: {
+          data: mockKubernetesPlrResponse.taskruns,
+        },
+      },
+      loaded: true,
+      responseError: '',
+      selectedClusterErrors: [],
+      clusters: [],
+      setSelectedCluster: () => {},
+    };
+    const { queryByTestId } = render(
+      <TektonResourcesContext.Provider value={mockContextData}>
+        <BrowserRouter>
+          <PipelineVisualization pipelineRun="pipeline-test-wbvtlk" />
+        </BrowserRouter>
+      </TektonResourcesContext.Provider>,
+    );
+    expect(
+      screen.getByText(mockKubernetesPlrResponse.pipelineruns[1].metadata.name),
+    ).toBeInTheDocument();
+    expect(screen.getByText('Pipeline Run')).toBeInTheDocument();
+
+    expect(screen.getByText('Back to PipelineRun list')).toBeInTheDocument();
+
+    expect(queryByTestId('pipelineRun-visualization')).not.toBeNull();
```

Comment:
can be consistent with the above assertions in same it block

```suggestion
    expect(queryByTestId('pipelineRun-visualization')).toBeInTheDocument();;
```